src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_CAPACITY_COEFFICIENT = 1.5;
+    private Object[] data;
+    private int size;
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= data.length) {
+            data = grow();
+        }
+        data[size] = value;
+        size += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size) {
```

Comment:
```suggestion
        if (index < 0 || index > size) {
```

Reply:
Can you please suggest the better way to apply your suggestions(when i'm studying and when developing a commercial priject):
- commit suggestion through github interface
- change code localy push and re-request review
Thank you.

Reply:
Always prefer a local resolving of suggestions (merge conflicts)